image_transport/test/test_subscriber.cpp

```diff
@@ -80,6 +80,25 @@ TEST_F(TestSubscriber, camera_sub_shutdown) {
   EXPECT_EQ(node_->get_node_graph_interface()->count_subscribers("camera/camera_info"), 0u);
 }
 
+TEST_F(TestSubscriber, callback_groups) {
```

Comment:
this test is not adding any checks

Reply:
Added two flags to make sure the callbacks are run. Does that work ?

Reply:
We have a multithreaded executor here with a reentrant callback group, so adding a sleep() to one of the callbacks here should not stop block the other callback. I'm not seeing that behavior, debugging.

Reply:
Fixed